routetable/route_table.go

```diff
@@ -230,7 +231,7 @@ func NewWithShims(
 	conntrack conntrackIface,
 	timeShim timeshim.Interface,
 	deviceRouteSourceAddress net.IP,
-	deviceRouteProtocol int,
+	deviceRouteProtocol, routeProtocol int,
```

Comment:
I don't think this new parameter gets used?

Given that we've got a route table for blackhole routes and one for non-blackhole I think each route table only needs one protocol parameter?  (and maybe we should rename the parameter to be more generic rather than the old deviceRouteProtocol?)